ganalytics.cpp

```diff
@@ -473,10 +501,12 @@ GAnalytics::GAnalytics(const QString &trackingID, QObject *parent)
 }
 
 /**
- * Destructor of class GAnalytics.
+ * Destructor of class GAnalytics. Ends the session.
  */
 GAnalytics::~GAnalytics()
 {
+    if (!trackingID().isEmpty())
+        endSession();
```

Comment:
Well, blame me for not documenting on this :worried:
We did not tie this library to a given way of saving the contents. This is a choice of the user here, the GAnalytics class is (de)serialisable using QDataStream.

Events are only sent in a given interval to Google, which happens asynchronously, this cannot be done in the constructor - if it would wait, some users might be disappointed about that behavior. 

In a normal Qt desktop application you usually save your window states using QMainWindow::saveState and then save this information somewhere you like (Windows Registry, Config Files, SQL Database). This process should also save the contents of the GAnalytics instance instead of relying on the class to save these values somewhere you may not like it :smiley: